src/coreclr/vm/jithelpers.cpp

```diff
@@ -3642,3 +3466,250 @@ void InitJITHelpers2()
     SetJitHelperFunction(CORINFO_HELP_INIT_PINVOKE_FRAME, (void *)GenerateInitPInvokeFrameHelper()->GetEntryPoint());
 #endif // TARGET_X86 || TARGET_ARM
 }
+
+
+#if defined(TARGET_32BIT) && !(defined(TARGET_X86) && defined(TARGET_WINDOWS))
+
+// Using the musttail attribute requires that we disble our protection against using return in the wrong place
+#ifdef return
+#undef return
+#endif
+
+extern "C" HCIMPL0(INT32, JIT_ThrowOverflow_RetInt)
+{
+    FCALL_CONTRACT;
+
+    FCThrow(kOverflowException);
```

Comment:
Could we directly inline them in JIT? there already are `CORINFO_HELP_OVERFLOW` and `CORINFO_HELP_THROWDIVZERO` exposed.

Reply:
Yes. This is just an experiment to see how musttail works, and if it can be used to help solve the HMF problem without changing the JIT/etc for this issue. Given the number of test failures, something clearly didn't go completely right with my change here, so I want to understand what I've missed.